code/datums/gameticker.dm

```diff
@@ -975,4 +979,62 @@ var/global/game_force_started = FALSE
 		src.creds = new /datum/crewCredits
 	return src.creds
 
+//% - this is the maximum possible chance with very few people on one server and a lot on another
+#define BASE_TOKEN_CHANCE 10
+// the ratio of players between this server and the highest one before drops start to occur
+//eg. 36 players on Morty, 60 on Sylvester = we start to drop tokens, at a very low chance
+#define RATIO_THRESHOLD 0.6
+
+/datum/controller/gameticker/proc/do_roundend_drops()
+	var/datum/game_server/self = global.game_servers.find_server(global.config.server_id)
+	var/is_nightshade = findtext(self.name, "nightshade")
+	var/datum/game_server/highest_pop = null
+	for (var/server_id in global.game_servers.servers)
+		var/datum/game_server/server = global.game_servers.servers[server_id]
+		var/other_is_nightshade = findtext(server.name, "nightshade")
+		if ((!!is_nightshade) ^ (!!other_is_nightshade)) //rare non-bitwise xor use case!!
+			continue
+		if (server.player_count > highest_pop?.player_count)
+			highest_pop = server
+
+	logTheThing(LOG_DEBUG, null, "Round end drops: we are [is_nightshade ? "" : "not"] nightshade, highest population server is [highest_pop?.name] with [highest_pop?.player_count] players")
+
+	//for when nightshade is winding down and we don't want to be giving people free tokens every round
+	if (highest_pop?.player_count <= 20)
+		logTheThing(LOG_DEBUG, null, "All linked servers below population threshold, no round end drops.")
+		return
+
+	var/pop_ratio = self.player_count / highest_pop.player_count
+	if (pop_ratio > RATIO_THRESHOLD)
+		logTheThing(LOG_DEBUG, null, "Population ratio with largest server: [pop_ratio], greater than threshold [RATIO_THRESHOLD], aborting end round drops.")
+		return // servers are reasonably balanced, no drops for u
+
+	//scale by round length, normalised at 50 minutes
+	var/length_ratio = min(src.round_elapsed_ticks/(50 MINUTES), 1)
+
+	var/actual_token_chance = BASE_TOKEN_CHANCE * ((RATIO_THRESHOLD - pop_ratio) / RATIO_THRESHOLD) * length_ratio
+	logTheThing(LOG_DEBUG, null, "Population ratio with largest server: [pop_ratio], starting end round drops with token chance [actual_token_chance]")
+
+	var/list/players = list()
+	for (var/mob/M as anything in mobs)
+		var/datum/player/player = M.mind.get_player()
+		if (player in players)
+			continue
+		players += player
+		//also scale by how long they've been in the round, no joining at the end just for the chance
+		var/time_ratio = player.current_playtime / ticker.round_elapsed_ticks
+		var/player_unique_chance = actual_token_chance * time_ratio
+		if (!prob(player_unique_chance))
+			continue //unlucky
+		if (player.get_antag_tokens() >= 1)
+			boutput(M, SPAN_BOLD("You would have received an antag token drop this round, but you already have one!"))
```

Comment:
I feel that giving players this information is just going to lead to them becoming frustrated.

Reply:
Orr making them less frustrated because they realise why they're not getting drops and spend their tokens accordingly.